src/data/astarte_device_sdk_lib.rs

```diff
@@ -18,19 +18,29 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
-use astarte_device_sdk::options::AstarteOptions;
+use astarte_device_sdk::options::{AstarteOptions, AstarteOptionsError};
 use astarte_device_sdk::types::AstarteType;
 use astarte_device_sdk::{
     registration, AstarteAggregate, AstarteDeviceDataEvent, AstarteDeviceSdk, AstarteError,
 };
 use async_trait::async_trait;
+use serde::Deserialize;
 
-use crate::data::Publisher;
+use crate::data::{Publisher, Subscriber};
 use crate::error::DeviceManagerError;
 use crate::repository::file_state_repository::FileStateRepository;
 use crate::repository::StateRepository;
 use crate::{get_hardware_id_from_dbus, DeviceManagerOptions};
 
+#[derive(Debug, Deserialize, Clone)]
+pub struct AstarteDeviceSdkConfigOptions {
+    pub realm: String,
+    pub device_id: Option<String>,
+    pub credentials_secret: Option<String>,
+    pub pairing_url: String,
+    pub pairing_token: Option<String>,
+}
+
 #[derive(Clone)]
 pub struct Astarte {
```

Comment:
I would take this chance to rename this struct in something more indicative of its use.
Something along the lines of: `AstarteDeviceSdkInstance`, `AstarteDevice`, `AstarteDeviceSdkLib`.